src/vma/dev/ring_simple.cpp

```diff
@@ -1278,7 +1278,6 @@ bool ring_simple::is_available_qp_wr(bool b_block)
 					if (ret <= 0) {
 						ring_logdbg("failed blocking on tx cq_mgr (errno=%d %m)", errno);
 						m_lock_ring_tx_buf_wait.unlock();
-						m_lock_ring_tx.lock();
```

Comment:
I believe we need to return/keep the lock here.
This function is called under a taken m_lock_ring_tx.lock(), so before return we need to make sure we hold the lock again.

again below... before return we need to hold lock().